src/components/AutoComplete/AutoComplete.tsx

```diff
@@ -0,0 +1,132 @@
+import { useCombobox } from "downshift";
+import { nanoid } from "nanoid";
+import React, { useState } from "react";
+import { useMergedProps } from "../../hooks/useMergedProps";
+import Field from "../internal/Field";
+import InternalAutoComplete from "../internal/InternalAutoComplete";
+import { defaultAutoCompleteProps } from "./defaultAutoCompleteProps";
+import { AutoCompleteProps } from "./types";
+
+function AutoComplete<Item>(props: AutoCompleteProps<Item | null>) {
+  const mergedProps = useMergedProps(props, defaultAutoCompleteProps);
+
+  const {
+    label,
+    innerRef,
+    error,
+    onChange,
+    onBlur,
+    onInputValueChange,
+    onSelectedItemChange,
+    onFocus,
+    disabled,
+    helpText,
+    items,
+    // getItems,
+    itemToString: itemToStringFn,
+    stateReducer,
+    placeholder,
+    isLoading,
+    itemsFooter,
+    listItem,
+    hideLabel,
+    testId,
+    value,
+    optional,
+    __internal__open,
+    __internal__highlightedIndex,
+    __internal__loading,
+    __internal__focus,
+  } = mergedProps;
+
+  const [auxId] = useState(() => `auto-complete-aux-${nanoid()}`);
```

Comment:
you could also use `useMemo` here. useState implies that it will change at some point in time.